homeassistant/components/overkiz/entity.py

```diff
@@ -48,6 +48,8 @@ def generate_device_info(self) -> DeviceInfo:
         # In this case, we use the base device url as the device identifier.
         if "#" in self.device_url and not self.device_url.endswith("#1"):
             # Only return the url of the base device, to inherit device name and model from parent device.
+            # Set the name in order to have different name for every sensor on the same device
+            self._attr_name = self.device.label
```

Comment:
Can you move this to the constructor? `generate_device_info` should not set a property.

Reply:
Sure, it's done.